.npm/postinstall.js

```diff
@@ -1,23 +1,55 @@
 require("process");
 const { spawn } = require("child_process");
+const { createWriteStream, unlink } = require("fs");
+const { get } = require("https");
 const { join } = require("path");
 
+const downloadPaths = {
+  'darwin': 'https://raw.githubusercontent.com/Arkweid/lefthook/master/.npm/bin/lefthook-mac',
+  'linux': 'https://raw.githubusercontent.com/Arkweid/lefthook/master/.npm/bin/lefthook-linux',
+  'win32': 'https://raw.githubusercontent.com/Arkweid/lefthook/master/.npm/bin/lefthook-win.exe'
+};
+
+function downloadBinary(os, callback) {
+  // construct single binary file path
+  const binaryPath = join(process.cwd(), 'lefthook');
```

Comment:
`process.cwd()`  is our current working directory. Consider some examples
`npm i @arkweid/lefhook` -- it will upload the binary to the project root
`cd src && npm i @arkweid/lefhook` -- it will upload the binary to 'src' 
We want the binary to be uploaded to `./node_modules/@arkweid/lefhook/...`

Reply:
Oh, whoops. I guess that was just the leftover `binaryPath` from my local tests. Will update immediately.

Reply:
`process.cwd()` should also give us the base path of the project, as the main function calls `process.chdir(process.env.INIT_CWD)`.

But if you want I can rewrite this, so that both the `downloadBinary` function takes `process.env.INIT_CWD` into consideration for the path, as well as call `process.chdir(process.env.INIT_CWD)` inside `installGitHooks`.

Reply:
We have two problems:
1. `process.env.INIT_CWD` refers to the current folder. (It could be a subfolder of the root folder.) So we need to look for `node_modules` recursively
2. Some older versions of `yarn`  don't set this variable at all

Reply:
1. No, you shouldn't search the folder node_modules at all (e.g., name of "node_modules" folder can be changed). There is simple and steady solution: https://github.com/levenkov/lefthook/pull/1/files#diff-5fe0be6c71fdcee66beee780a898073fR167

1. Download it in preinstall hook.
2. preinstall hook executed in the package directory.
3. It's always true, that if you placed downloaded binary in `./bin` in the preinstall hook, this binary after install stage will be in the right place (usually here: `<project dir>/node_modules/@arkweid/lefthook/bin/...`).

I almost done the work, but was overwhelmed by the deadline :)

Reply:
I was under the impression that the `INIT_CWD` is set to the base path of the newly installed package (see https://github.com/yarnpkg/yarn/blob/cfb560e1b2e37ed5cfe5d2374a806ac3ae512c11/src/util/execute-lifecycle-script.js#L40 similar for `npm`), hence `INIT_CWD` for `lefthook` would always be `<project-dir>/node_modules/@Arkweid/lefthook`.

(Also the change was (according to GitHub) over a year ago, so I would assume most users have a current enough `yarn`/`npm` installation)



So tl;dr: As long as we don't `process.chdir` or `cd <somewhere> && ...` in the postinstall script, we should be able to assume we are in `<project-dir>/node_modules/@Arkweid/lefthook`, no?

Reply:
> (Also the change was (according to GitHub) over a year ago, so I would assume most users have a current enough yarn/npm installation)

It's better to support as earlier as you can version of node. It's common place to support >=6.0.

INIT_CWD is working after 8.

Reply:
@levenkov @HellSquirrel 

I honestly don't know how to fix this `INIT_CWD` issue. I tried to use the `./dir` approach (i.e. `const path = process.env.INIT_CWD || path.resolve(".");`), but in my locally installation tests, this always resolved to the path the script resides in (i.e. `<path>/lefthook/.npm`, instead of `<path>/tmp` where I called `npm install ../lefthook/.npm` from).

Maybe you can review the latest PR version and give me some pointers.

Reply:
@fleischie I'll see that in 1-2 days. Maybe I'll try to change the approach in your code to mine and suggest small patch. I don't claim to replace all your work, just this problem.